File: 프로그래머스/1/42748. K번째수/K번째수.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>

using namespace std;

vector<int> solution(vector<int> array, vector<vector<int>> commands)
{
    vector<int> answer;
    
    for(int i = 0; i < commands.size(); i++)
    {
        int startIndex = commands[i][0];
        int endIndex = commands[i][1];
        int returnIndex = commands[i][2];
        
        vector<int> value;
        for(int j = startIndex; j <= endIndex; j++)
        {
            int target = array[j - 1];
            value.push_back(target);
        }
        
        sort(value.begin(), value.end());
    
        answer.push_back(value[returnIndex - 1]);
    }
    
    return answer;
}

File: 프로그래머스/1/42748. K번째수/K번째수.cs
using System;

public class Solution 
{
    public int[] solution(int[] array, int[,] commands)
    {
       int[] answer = new int[commands.GetLength(0)];
        
        for(int i = 0; i < commands.GetLength(0); i++)
        {
            var startIndex = commands[i, 0];
            var endIndex = commands[i, 1];
            var targetIndex = commands[i, 2];
            
            var tempArray = new int[endIndex - startIndex + 1];
            for(int j = 0; j < tempArray.Length; j++)
            {
                tempArray[j] = array[startIndex - 1];
                startIndex++;
            }
            
            Array.Sort(tempArray);
            
            answer[i] = tempArray[targetIndex - 1];
        }
        
        return answer;
    }
}